atta/analyzer/generic.py

```diff
@@ -8,199 +8,65 @@
+    return df_selected
+
+def add_cat_title(df):
     df['Title_Categories'] = df['Title'].apply(cat_title)
-    df
-
-
-    # In[32]:
-
-
-    pattern_nobody = "(?i)無|0|沒有人|nobody|nan|自由業|none|self-employed"
+    return df
 
-    pattern = '家|自'
 
-    # df2 =pd.DataFrame( {'A':[1,2,3],
-    #      'B':[4,5,6],'C':[7,8,9]})
-    # df3 = df2[['A','B']]
-    # df3
 
-    # Without a job: See from Organization
+def plot_counts(df,year):
+    cols = df.keys().tolist()
+    bar_cols = [cols[0],cols[1],cols[4]]  # change here to include more columns for count plots
+    for col in bar_cols:
+        fig, ax = plt.subplots(figsize=(12, 8))
+        plt.title(str(col)+' of the Attendees at PyCon Taiwan in ' + str(year))
+        ax.set_xticklabels(str(col), rotation=0, fontdict={"fontsize": '8'})
+        ax.set_xlabel(xlabel=str(col))
+        ax.set_ylabel(ylabel="Counts")
+        sns.set(font_scale=2)
+        sns.countplot(x = str(col), data = df)
+        # directory = Tk.tk() # Something about
+        plt.savefig(str(col) + str(year) + '.jpg')
 
-    # Nan: 32,71,101,104,325,569
-    # none: 194, 274, 276,301,375,426, 498
-    # 不告訴你： 9,263,560
-    # 家裡個人接案：435,436,465,
-    # 560 : 表情符號...
+# def plot_accumulated_count(df,year):
 
-    # 488是駭客！
+# def save_df():
+#     df.to_pickle(str(year)+"_analyzed_data")  ## need to refine directory
 
-    df[['Organization','Title']].iloc[9]
 
+def go(csvfile):
 
-    # ## Plot title categories
-
-
-
-    df['Title_Categories'].value_counts('Job Seeker')
-
-
-
-
-    fig, ax = plt.subplots()
-    fig.set_size_inches(11.7, 8.27)
-    ax.set_xticklabels('1_Title',rotation=90,fontdict= {"fontsize": '16'})
-    ax.set_xlabel(xlabel="Job Title")
-    ax.set_ylabel(ylabel= "Counts")
-    sns.set(font_scale = 2)
-
-    sns.countplot(x = 'Title_Categories', data = df, ax = ax,
-                  order = ['Engineer', 'Student', 'Developer', 'Manager', 'Researcher',
-           'Data Scientist', 'Head', 'Professor', 'Job Seeker', 'Consultant',
-           'Graphics', 'Others'])
-
-
-    fig, ax = plt.subplots()
-    fig.set_size_inches(11.7, 8.27)
-    ax.set_xticklabels('1_Title',rotation=45,fontdict= {"fontsize": '16'})
-    ax.set_xlabel(xlabel="Job Title")
-    ax.set_ylabel(ylabel= "Counts")
-    sns.set(font_scale = 2)
-
-    sns.barplot(x = '1_Title', y = "2_Counts", data = title_cat_counts, ax = ax)
-
-    plt.show()
+    '''Get file. Replace hardcoded line with CLI input'''
+    # welcome user, ask year of data
+    year = welcome_ask_year()
+    # read csv as pandas dataframe
+    df = pd.read_csv(csvfile)
+    # Print the keys as bug messages
+    logging.debug(df.keys())
+    # select necessary columns from df
+    df = select_column(df)
+    df = add_cat_title(df)
+    plot_counts(df,year)
+    print('Go process finished completely. Find results in the same directory as the csv file.')
 
 def fake_go():
-    print('fake_go is executed.')
+    print('fake_go is executed.')
```

Comment:
A common recommended coding style is to add an new line at the end of the file. Please add it. Thanks.